src/index.js

```diff
@@ -1,113 +1,79 @@
-        exec(cmd, function (err) {
-          if (!err) {
-            resolve()
-          } else {
-            reject(new Error('Unable to compress app.'))
-          }
-        })
-      }
-    })
-  }
+export function normalizeOptions(opts = {}) {
+  let pkg = loadPackageJson()
 
-  // Create new release with zip as asset.
-  release () {
-    let self = this
+  if (!opts.repo) opts.repo = getRepo(pkg)
+  if (!opts.tag) opts.tag = getTag(pkg)
+  if (!opts.name) opts.name = opts.tag
+  if (!opts.output) opts.output = opts.app
 
-    return new Promise(function (resolve, reject) {
-      publishRelease({
-        token: self.opts.token,
-        owner: self.opts.repo.split('/')[0],
-        repo: self.opts.repo.split('/')[1],
-        tag: self.opts.tag,
-        name: self.opts.name,
-        assets: self.opts.output
-      }, function (err, release) {
-        if (!err) {
-          got(release.assets_url).then(function (res) {
-            var jsonBody = JSON.parse(res.body)
-            self._releaseUrl = jsonBody[0].browser_download_url
-            resolve()
-          })
-        } else {
-          reject(new Error('Unable to create a new release on GitHub.'))
-        }
-      })
-    })
-  }
+  return opts
+}
 
-  // Update auto_update.json file with latest url.
-  updateUrl () {
-    let self = this
-    return new Promise(function (resolve) {
-      loadJsonFile('./auto_updater.json').then(function (content) {
-        content.url = self._releaseUrl
-        writeJsonFile('./auto_updater.json', content).then(function () {
-          resolve()
-        })
-      })
-      .catch(function (err) {
-        resolve()
-      })
-    })
-  }
+export function compress({ app, output }) {
+  if (!Array.isArray(app)) app = app.replace(/ /g, '').split(',')
+  if (!Array.isArray(output)) output = output.replace(/ /g, '').split(',')
 
-  // Load package.json
-  _loadPackageJson () {
-    try {
-      return loadJsonFile.sync('./package.json')
-    } catch (err) {
-      return
-    }
+  if (app.length !== output.length) {
+    return Promise.reject(new Error('Output length does not match app length'))
   }
 
-  // Get repo from package.json
-  _getRepo () {
-    let pkg = this._loadPackageJson()
-
-    let url = pkg.repository.url.split('/')
-    return url[3] + '/' + url[4].replace(/\.[^/.]+$/, '')
-  }
+  return Promise.resolve(app).map((item, index) => {
+    let outputZip = (path.extname(output[i]) === '.zip') ? output[i] : output[i] + '.zip'
+    let cmd = `ditto -c -k --sequesterRsrc --keepParent ${item} ${outputZip}`
 
-  // Get tag (version) from package.json
-  _getTag () {
-    let pkg = this._loadPackageJson()
+    return execAsync(cmd).catch(err => {
+      throw new Error('Unable to compress app.')
+    })
+  })
+}
 
-    let version = pkg.version
-    return 'v' + version
-  }
+export function release({ token, repo, tag, name, output }) {
+  return publishReleaseAsync({
```

Comment:
I was able to get it working over here: https://github.com/thejameskyle/__electron-release-test

I need to clean something up though.